hie-brittany/Haskell/Ide/BrittanyPlugin.hs

```diff
@@ -0,0 +1,92 @@
+{-# OPTIONS_GHC -fno-warn-partial-type-signatures #-}
+
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE FlexibleContexts      #-}
+{-# LANGUAGE GADTs                 #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+{-# LANGUAGE ScopedTypeVariables   #-}
+{-# LANGUAGE LambdaCase            #-}
+module Haskell.Ide.BrittanyPlugin where
+
+import           Data.Aeson
+import           Data.Monoid
+import qualified Data.Text as T
+import           Data.Vinyl
+import           Haskell.Ide.Engine.PluginDescriptor
+import           Haskell.Ide.Engine.PluginUtils
+import           Control.Monad.IO.Class
+import           Language.Haskell.Brittany
+import           Language.Haskell.Brittany.Types
+import           Language.Haskell.Brittany.Config
+import qualified DynFlags as GHC
+import qualified GHC.LanguageExtensions.Type as GHC
+import qualified Data.Text.Lazy as TextL
+import qualified GhcMod.Utils as GM
+import qualified Language.Haskell.LSP.TH.DataTypesJSON as J
+import           Control.Lens
+
+
+brittanyCmd :: TextDocumentIdentifier -> Maybe Range -> IdeM (IdeResponse [J.TextEdit])
+brittanyCmd tdi range =
+  pluginGetFile "genapplicative: " (tdi ^. J.uri) $ \file -> do
```

Comment:
There are a number of strings for `pluginGetFile` that do not actually match. Not sure how we can test for them though. Perhaps better to leave them out and let the exception location guide us?